lib/engine/game/g_1825/game.rb

```diff
@@ -704,6 +705,18 @@ def tile_color_valid_for_phase?(tile, phase_color_cache: nil)
           phase_color_cache.include?(tile.color)
         end
 
+        def location_name(coord)
+          unless @location_names
+            @location_names = LOCATION_NAMES.dup
+            if optional_rules.include?(:unit_4)
+              @location_names['A5'] = 'Inverness'
+              @location_names['C3'] = 'Fort William'
+              @location_names.delete('B8')
+            end
+          end
+          @location_names[coord]
+        end
+
```

Comment:
@locations_name || = ....

Reply:
I'm sorry, I'm not really clear on what you're suggesting here, beyond "implement this differently". I copied what 1824 does. Is there another game I should emulate, please?

Reply:
Assigning a "caching" variable has a pretty standard usage pattern on 18xx.games. Look at the definition of bank_by_options in 1825 for an example. Or just git grep for '||='.

Reply:
Given what you are doing, this is fine.

I would do it like this (mostly just personal preference, same effect as what you have):
```
def location_name(coord)
  @locations_names ||= game_location_names
  @locations_names[coord]
end

def game_location_names
  locations = LOCATION_NAMES.dup
  if optional_rules.include?(:unit_4)
 ...
  end
end
```

Reply:
I haven't forgotten about this, I'll do it when I've got some spare time.

Reply:
Like I said, I think what you have is fine. If you don't have time, don't worry about it.